### DISCUSSION TITLE
Office live

### DISCUSSION THREAD

**Discussion by kollbrt** [SafeExamBrowser/seb-win-refactoring#228] [General]
We woult like to design an exam including opening a word-file out of open olat with the tool from office live. We tried to allow all websites connectet with office live but we still don't manage to open the wordfile during a running SEB session. What should we do? It always shows, that the process is blocked and the site doesn't open.  Thank's for your help.

**Comment by dbuechel** [comment]
First I'd recommend turning the URL filter off to see whether it works without the filter. Secondly, if you'd like to use e.g. Word during a session, you'd need to add it to the list of permitted applications in the configuration: https://safeexambrowser.org/windows/win_usermanual_en.html#PermittedProcessesSection

**Reply by kollbrt** [reply]
I tried it already without the URL Filter. And do you think I need to permit the word application although it is just the office.live version directly online?

**Reply by dbuechel** [reply]
Aha, I misunderstood you then. No, if you use the online version you don't need to add Word to the permitted applications. Could you share a printscreen or photo of the exact error message you get when trying to open the document?

**Reply by kollbrt** [reply]
Here is the picture of the error message, after I clicked on the button in Olat to open the document on microsoft office online.

![IMG_20211023_185541](https://user-images.githubusercontent.com/74318647/138565152-e1d8a2fd-e77e-43b4-a2f8-b639b9f5c0ae.jpg)

**Comment by dbuechel** [comment]
Hm, it appears that OLAT is blocking something here, not SEB. SEB would show an error message if the URL filter were active. Could you please provide the client log file of the affected session? The logs for version 3.x can be found under `%LocalAppData%\SafeExamBrowser\Logs`.

**Reply by kollbrt** [reply]
Here they are: 

[2021-10-25_11h24m00s_Runtime.log](https://github.com/SafeExamBrowser/seb-win-refactoring/files/7408894/2021-10-25_11h24m00s_Runtime.log)
[2021-10-25_11h24m00s_Client.log](https://github.com/SafeExamBrowser/seb-win-refactoring/files/7408895/2021-10-25_11h24m00s_Client.log)
[2021-10-25_11h24m00s_Browser.log](https://github.com/SafeExamBrowser/seb-win-refactoring/files/7408896/2021-10-25_11h24m00s_Browser.log)

**Comment by dbuechel** [comment]
Yes indeed, as the log shows, SEB isn't blocking anything here. It appears that there is something in OLAT itself blocking your use case, does that LMS maybe have a URL filtering mechanism or something alike as well?

**Reply by dbuechel** [reply]
You might want to allow the logging of URLs and opening of new windows temporarily in your configuration to debug the issue: https://safeexambrowser.org/windows/win_usermanual_en.html#BrowserPane (Links requested to be opened in new browser window -> "open in new window" and Show URLS -> "before title").

**Reply by kollbrt** [reply]
I don't know if there is an URL Filtering. But the strange thing is, that without the SEB, the same process is perfectly working out of Olat. Only if SEB is running, I'm not able to start office.live

**Reply by kollbrt** [reply]
> 
> 
> You might want to allow the logging of URLs and opening of new windows temporarily in your configuration to debug the issue: https://safeexambrowser.org/windows/win_usermanual_en.html#BrowserPane (Links requested to be opened in new browser window -> "open in new window" and Show URLS -> "before title").

I tried also your advice with "open in new window" and show URLs. That didn't work.

**Reply by danschlet** [reply]
Look into if third party cookies are used. I think there was an issue reported recently related to that. I think neither SEB for Windows nor macOS/iOS supports those currently.

**Reply by kollbrt** [reply]
Where can I look if there are third party cookies used?

**Reply by danschlet** [reply]
Is this a built-in feature of OpenOLAT, to open a word file with that tool from office live? If yes, then you need to ask OpenOLAT support how to make this run with SEB. They would also know if this may be an issue related to third party cookies.

As @dbuechel seems to have found out, it's an issue with the web application or a compatibility issue with SEB. **As we mention on our support page, ALWAYS first ask the provider of your exam system to help you solve problems related to that particular exam system/LMS**. OpenOLAT is contributor to the SEB Alliance, so we can work together with them to help solving the issue.

We are only able to give general support for Safe Exam Browser. SEB can be used with various different exam systems, many of which we don't even know. We only provide the Safe Exam Browser kiosk browser software itself.

**Reply by kollbrt** [reply]
Thank you both! So we will fix this with frentix from open olat. I didn't realize that the problem lies with olat, because without the SEB, it worked well to open a document with the built-in feature to open word files online.

**Comment by dbuechel** [comment]
@kollbrt Can you please provide the log files of the session where you have these two settings enabled?

**Reply by kollbrt** [reply]
[2021-10-25_11h42m10s_Runtime.log](https://github.com/SafeExamBrowser/seb-win-refactoring/files/7409041/2021-10-25_11h42m10s_Runtime.log)
[2021-10-25_11h42m10s_Browser.log](https://github.com/SafeExamBrowser/seb-win-refactoring/files/7409042/2021-10-25_11h42m10s_Browser.log)
[2021-10-25_11h42m10s_Client.log](https://github.com/SafeExamBrowser/seb-win-refactoring/files/7409043/2021-10-25_11h42m10s_Client.log)

**Comment by dbuechel** [comment]
The URLs still aren't logged, but I guess they won't provide much more insight into the issue. It is as suspected, OLAT or a component of it blocks the navigation to office live:

```
2021-10-25 11:42:28.401 [28] - DEBUG: [Browser Instance #1] Navigated.
2021-10-25 11:42:31.644 [28] - DEBUG: [Browser Instance #1] Forwarding request to open new window...
2021-10-25 11:42:31.646 [28] - INFO: [BrowserApplication] Received request to create new instance for 'about:blank#blocked'...
```

**Comment by dbuechel** [comment]
Have you verified that the issue does not appear in Google Chrome or Chromium?

**Reply by dbuechel** [reply]
You can also verify whether this might be an issue of the browser engine by trying to execute your use case in the CEF sample application: https://cef-builds.spotifycdn.com/index.html#windows64

**Reply by kollbrt** [reply]
How can I define, that Google Chrome should be used with my SEB settings? I didn't try that yet.

**Reply by dbuechel** [reply]
You can't use SEB settings with Chrome. What I'm referring to is opening your start URL in Chrome and trying your use case in the browser.

**Comment by uhensler** [comment]
We have investigated the problem from OpenOlat's perspective and have found the following.

The problem has nothing to do with Office for the web (= office.live), it is an extraordinary behavior in the execution of Javascript.

OpenOlat opens a new window/tab with JavaScript. This is the corresponding snippet.

```
var targetWindow = window.open("","_blank");
...
targetWindow.location.href = '<the real url>';
```

Unlike all other browsers, the SEB returns null in method `window.open(...)'`. As a result, setting the URL in the second statement does not work.

From our point of view, this is a bug in the SEB. Can you confirm the detected behavior of `window.open(...)`? Is it intentional? Can it be disabled by a configuration?

It is surprising that according to BBW it worked on 09/16/2021, but on 10/21/2021 it did not anymore. The relevant code in OpenOlat has not changed for two years.

**Reply by dbuechel** [reply]
I think we'll need to investigate this further on our side. Apparently, it appears to be working correctly in the CEF sample application (https://cef-builds.spotifycdn.com/index.html#windows64) so there must be an issue in either CefSharp or our code.

**Reply by dbuechel** [reply]
Please refer to #240 for further discussion.

**Reply by dbuechel** [reply]
@uhensler It turned out that we can't fix the issue for now, see #240 for the details.
